src/Silex/CallbackResolver.php

```diff
@@ -33,40 +33,59 @@ public function __construct(Container $app)
      */
     public function isValid($name)
     {
-        return is_string($name) && preg_match(static::SERVICE_PATTERN, $name);
+        return is_string($name) && (preg_match(static::SERVICE_PATTERN, $name) || isset($this->app[$name]));
     }
 
     /**
      * Returns a callable given its string representation.
      *
      * @param string $name
      *
-     * @return array A callable array
+     * @return callable A callable value
      *
      * @throws \InvalidArgumentException In case the method does not exist.
      */
     public function convertCallback($name)
     {
-        list($service, $method) = explode(':', $name, 2);
+        if (preg_match(static::SERVICE_PATTERN, $name)) {
+            list($service, $method) = explode(':', $name, 2);
 
-        if (!isset($this->app[$service])) {
-            throw new \InvalidArgumentException(sprintf('Service "%s" does not exist.', $service));
-        }
+            if (!isset($this->app[$service])) {
+                $this->throwServiceNotExists($service);
+            }
+
+            return array($this->app[$service], $method);
+        } else {
+            if (!isset($this->app[$name])) {
+                $this->throwServiceNotExists($name);
+            }
+
+            $callback = $this->app[$name];
+
+            if (!is_callable($callback)) {
+                throw new \InvalidArgumentException(sprintf('Service "%s" is not callable.', $name));
+            }
 
-        return array($this->app[$service], $method);
+            return $callback;
+        }
     }
 
     /**
      * Returns a callable given its string representation if it is a valid service method.
      *
      * @param string $name
      *
-     * @return array A callable array
+     * @return string|callable A callable value or the string passed in
      *
      * @throws \InvalidArgumentException In case the method does not exist.
      */
     public function resolveCallback($name)
     {
         return $this->isValid($name) ? $this->convertCallback($name) : $name;
     }
+
+    private function throwServiceNotExists($service)
+    {
+        throw new \InvalidArgumentException(sprintf('Service "%s" does not exist.', $service));
+    }
```

Comment:
I didn't had a look at the code, just the coding standard but what about inlining this method? I think having the same exception duplicated twice is fine.